providers/aws/aws_provider.py

```diff
@@ -115,11 +126,40 @@ def provider_set_session(
     current_audit_info.audited_account = caller_identity["Account"]
     current_audit_info.audited_partition = arnparse(caller_identity["Arn"]).partition
 
+    logger.info("Checking if organizations role assumption is needed ...")
+    if organizations_role_arn:
+        current_audit_info.assumed_role_info.role_arn = organizations_role_arn
+        current_audit_info.assumed_role_info.session_duration = input_session_duration
+
+        # Check if role arn is valid
+        try:
+            # this returns the arn already parsed, calls arnparse, into a dict to be used when it is needed to access its fields
+            role_arn_parsed = arn_parsing(current_audit_info.assumed_role_info.role_arn)
+
+        except Exception as error:
+            logger.critical(f"{error.__class__.__name__} -- {error}")
+            sys.exit()
+
+        else:
+            logger.info(
+                f"Getting organizations metadata for account {organizations_role_arn}"
+            )
+            assumed_credentials = assume_role(current_audit_info)
+            current_audit_info.organizations_metadata = get_organizations_metadata(
+                assumed_credentials
+            )
+            logger.info(f"Organizations metadata retrieved")
+
     logger.info("Checking if role assumption is needed ...")
-    if current_audit_info.assumed_role_info.role_arn:
+    if input_role:
+        current_audit_info.assumed_role_info.role_arn = input_role
+        current_audit_info.assumed_role_info.session_duration = input_session_duration
+        current_audit_info.assumed_role_info.external_id = input_external_id
+
         # Check if role arn is valid
         try:
             # this returns the arn already parsed, calls arnparse, into a dict to be used when it is needed to access its fields
+            print(current_audit_info.assumed_role_info)
```

Comment:
Do we need this `print` here? Maybe little debugging code forgotten?
Check it out